File: CsharpTDD/CsharpTDD/Money.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CsharpTDD
{
    internal class Money : IExpression
    {
        public int amount;

        public string currency;

        public Money Times(int multiplier)
        {
            return new Money(amount * multiplier, currency);
        }

        public string Currency()
        {
            return this.currency;
        }

        public Money(int amount, string currency)
        {
            this.amount = amount;
            this.currency = currency;
        }

        public static Money Dollor(int amout)
        {
            return new Money(amout, "USD");
        }

        public static Money Franc(int amout)
        {
            return new Money(amout, "CHF");
        }

        public override bool Equals(object obj)
        {
            Money money = (Money)obj;
            return amount == money.amount && Currency().Equals(money.Currency());
        }

        public IExpression Plus(Money addend)
        {
            return new Sum(this, addend);
        }

        public Money Reduce(string to) 
        {
            return this;
        }
    }
}

File: JavaTdd/src/main/java/com/sangkon/Money.java
package com.sangkon;

public class Money {
    protected int amount;
    protected String currnecy;

    public Money(int amount, String currnecy) {
        this.amount = amount;
        this.currnecy = currnecy;
    }


    public static Money dollar(int amount) {
        return new Money(amount, "USD");
    }

    public static Money franc(int amount) {
        return new Money(amount, "CHF");
    }


    public Money times(int multiplier) {
        return new Money(amount * multiplier, currnecy);
    }

    public String currency() {
        return currnecy;
    }

    @Override
    public boolean equals(Object o) {
        Money money = (Money) o;
        return amount == money.amount && currency().equals(money.currnecy);
    }

    @Override
    public String toString() {
        return "Money{" +
                "amount=" + amount +
                ", currnecy='" + currnecy + '\'' +
                '}';
    }
}
